python/dgl/view.py

```diff
@@ -277,7 +277,8 @@ def __getitem__(self, key):
 
     def __call__(self, ntype=None):
         """Return the nodes."""
-        return F.arange(0, self._graph.number_of_nodes(ntype))
+        return F.arange(0, self._graph.number_of_nodes(ntype),
+                        dtype=F.data_type_dict[self._graph._graph.dtype])
```

Comment:
This makes me think whether we should directly return backend dtype object in `g.idtype`. This would make everything much easier.

Reply:
There are numpy dtype, backend dtype and string literal dtype here. Conversions are needed at multiple places. String literal dtype is easiest for conversion and comparison. Therefore I choose str finally.

Reply:
I get there are multiple options here, but which one is more convenient for the common cases? That should to be backend dtype right?

Reply:
It's not convenient to convert backend dtype to np dtype(to initialize dgl ndarray, which is widely used). It's something like `getattr(np, F.reverse_data_dtype[dtype])`. I still prefer using string to represent dtypes

Reply:
> It's not convenient to convert backend dtype to np dtype(to initialize dgl ndarray, which is widely used). It's something like `getattr(np, F.reverse_data_dtype[dtype])`. I still prefer using string to represent dtypes

Is that visible to user? Can we allow initialize dgl ndarray with backend dtype?

Reply:
Which api are you referring to(visible to user)? Why do you think backend dtype is better? How about another alias?